src/main/java/org/folio/services/transactions/BatchTransactionService.java

```diff
@@ -70,4 +80,35 @@ private Future<Void> updateTransaction(Transaction transaction, RequestContext r
       .withTransactionsToUpdate(singletonList(transaction));
     return processBatch(batch, requestContext);
   }
+
+  private Future<Void> checkDeletions(Batch batch, RequestContext requestContext) {
+    if (batch.getIdsOfTransactionsToDelete().isEmpty()) {
+      return succeededFuture();
+    }
+    return anyConnectedToInvoice(batch.getIdsOfTransactionsToDelete(), requestContext)
+      .map(connected -> {
+        if (FALSE.equals(connected)) {
+          return null;
+        }
+        logger.warn("validateDeletion:: Tried to delete transactions but one is connected to an invoice, ids={}",
+          batch.getIdsOfTransactionsToDelete());
+        throw new HttpException(422, DELETE_CONNECTED_TO_INVOICE.toError());
+      });
+  }
+
+  private Future<Boolean> anyConnectedToInvoice(List<String> ids, RequestContext requestContext) {
+    // We want to know if the order with the given encumbrance is connected to an invoice.
```

Comment:
probably we can add extend this comment with example where this check is required (open order -> approve invoice -> unopen order -> remove fund distrib line)